server/signedcontainer/verify.go

```diff
@@ -0,0 +1,130 @@
+package signedcontainer
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/GoogleCloudPlatform/confidential-space/server/signedcontainer/internal/convert"
+	"github.com/tink-crypto/tink-go/v2/keyset"
+	tinksig "github.com/tink-crypto/tink-go/v2/signature"
+)
+
+type ImageSignature struct {
+	Payload   []byte
+	Signature []byte
+	Alg       SigningAlgorithm
+}
+
+const maxSignatureCount = 300
+
+// Verify attempts to verify the provided signatures with imageDigest and returns:
+//   - []*ImageSignature - list of successfully verified signatures
+//   - []error           - (nonblocking) list of errors from unsuccessful verifications
```

Comment:
It's still a bit confusing if `[]error` is returned along with error. The intention of having this field was to return partial error details related to signature verification for callers to log/analyze. To open source this functionality, my advice is to consolidate verified signatures + partial errors into a new struct `VerifyResults`.
e.g.,
```
type VerifyResult {
  ValidSigs []*ImageSignature
  PartialErrors []error
}

func Verify(imageDigest string, signatures []*ImageSignature) (*VerifyResult, error) {
```

Reply:
Done. I was trying to figure out how to make the function signature less confusing, thanks!